File: algorithm/sort/js/select.js
let arr = [1, 3, 5, 79, 0, 2, 88, 66, 37, 21]

function selectionSort(arr) {
  let len = arr.length;
  for (let j = 0; j < len - 1; j++) {
    let minimum = j;
    for (let i = j + 1; i < len; i++) {
      if (arr[i] < arr[minimum]) {
        minimum = i;
      }
    }
    let temp = arr[j];
    arr[j] = arr[minimum];
    arr[minimum] = temp;
  }
  return arr;
}
console.log(selectionSort(arr)); // [0, 1, 2, 3, 5, 21, 37, 66, 79, 88]

File: algorithm/sort/rust/select.rs
fn slect_sort(arr: &mut [i32; 10]) {
  for i in 0..arr.len() {
    let mut minimum = i;
    for j in i..arr.len() {
      if arr[j] < arr[minimum] {
        minimum = j;
      }
    }
    arr.swap(i, minimum);
  }
}

fn main() {
  let mut arr: [i32; 10] = [1, 3, 5, 79, 0, 2, 88, 66, 37, 21];
  slect_sort(&mut arr);
  println!("{:?}", arr);
}
